src/components/movement-sensor/client.ts

```diff
@@ -197,8 +194,37 @@ export class MovementSensorClient implements MovementSensor {
     return acc.toObject();
   }
 
-  getReadings(extra = {}) {
-    return this.sensorclient.getReadings(extra);
+  async getReadings(extra = {}) {
+    const readings: {
+      position?: MovementSensorPosition;
+      linear_velocity?: Vector3;
+      angular_velocity?: Vector3;
+      linear_acceleration?: Vector3;
+      compass_heading?: number;
+      orientation?: Orientation;
+    } = {};
+    const readingFunctions: Record<keyof typeof readings, CallableFunction> = {
+      position: this.getPosition.bind(this),
+      linear_velocity: this.getLinearVelocity.bind(this),
+      angular_velocity: this.getAngularVelocity.bind(this),
+      linear_acceleration: this.getLinearAcceleration.bind(this),
+      compass_heading: this.getCompassHeading.bind(this),
+      orientation: this.getOrientation.bind(this),
+    };
+    /* eslint-disable no-await-in-loop */
+    for (const [field, func] of Object.entries(readingFunctions)) {
+      try {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        readings[field as keyof typeof readings] = await func(extra);
+      } catch (error) {
+        if (!(error as Error).message.includes('Unimplemented')) {
+          throw error;
+        }
+      }
+    }
+    /* eslint-enable no-await-in-loop */
```

Comment:
I'd really like to get rid of the no-unsafe-assignment issue... the no-await one isn't as big a of deal. but in any case we can just have both ignores on the same line like so:

```suggestion
    for (const [field, func] of Object.entries(readingFunctions)) {
      try {
        // eslint-disable-next-line no-await-in-loop,@typescript-eslint/no-unsafe-assignment
        readings[field as keyof typeof readings] = await func(extra);
      } catch (error) {
        if (!(error as Error).message.includes('Unimplemented')) {
          throw error;
        }
      }
    }
```